ovs/lib/storagerouter.py

```diff
@@ -1021,13 +1031,15 @@ def remove_storagedriver(storagedriver_guid, offline_storage_router_guids=None):
                     StorageRouterController._logger.exception('Remove Storage Driver - Guid {0} - Destroying filesystem and erasing node configs failed'.format(storage_driver.guid))
                     errors_found = True
             try:
-                if ServiceManager.has_service(albaproxy_service, client=client):
-                    StorageRouterController._logger.info('Remove Storage Driver - Guid {0} - Stopping service {1}'.format(storage_driver.guid, albaproxy_service))
-                    ServiceManager.stop_service(albaproxy_service, client=client)
-                    StorageRouterController._logger.info('Remove Storage Driver - Guid {0} - Removing service {1}'.format(storage_driver.guid, albaproxy_service))
-                    ServiceManager.remove_service(albaproxy_service, client=client)
+                for proxy in storage_driver.alba_proxies:
+                    service_name = proxy.service.name
+                    if ServiceManager.has_service(service_name, client=client):
+                        StorageRouterController._logger.info('Remove Storage Driver - Guid {0} - Stopping service {1}'.format(storage_driver.guid, service_name))
+                        ServiceManager.stop_service(service_name, client=client)
+                        StorageRouterController._logger.info('Remove Storage Driver - Guid {0} - Removing service {1}'.format(storage_driver.guid, service_name))
+                        ServiceManager.remove_service(service_name, client=client)
             except Exception:
-                StorageRouterController._logger.exception('Remove Storage Driver - Guid {0} - Disabling/stopping service {1} failed'.format(storage_driver.guid, albaproxy_service))
+                StorageRouterController._logger.exception('Remove Storage Driver - Guid {0} - Disabling/stopping services failed'.format(storage_driver.guid))
```

Comment:
Maybe also include the name of the service?
This information could be fetched from the DAL in the worst case but could be useful to have it logged.